New package: Feature graph (#9465)
CGAL/cgal: Feature_graph/doc/Feature_graph/PackageDescription.txt

// PRETTY PACKAGE NAME should equal the project title in Doxyfile.in

/// \defgroup PkgFeatureGraphRef Reference Manual

/// \defgroup PkgFeatureGraphConcepts Concepts
/// \ingroup PkgFeatureGraphRef

/// \defgroup PkgFeatureGraphDetector Feature Graph Detector
/// \ingroup PkgFeatureGraphRef

> /// The detectors in this group allows to extract the feature graph from different types of surfaces.

Reviewer: ```suggestion
/// The detectors in this group allow to extract the feature graph from different types of surfaces.
```

/// \defgroup PkgFeatureGraphSharpnessEstimator Sharpness Estimator
/// \ingroup PkgFeatureGraphRef

> /// The detectors in this group allows to estimate the sharpness on surfaces.

Reviewer: ```suggestion
/// The detectors in this group allow to estimate the sharpness on surfaces.
```

/// \defgroup PkgFeatureGraphNormalEstimator Normal Estimator
/// \ingroup PkgFeatureGraphRef

> /// The detectors in this group allows to estimate the normal on surfaces.

Reviewer: ```suggestion
/// \defgroup PkgFeatureGraphNormalEstimator Normal Estimator
/// \ingroup PkgFeatureGraphRef
/// The detectors in this group allow to estimate the normal on surfaces.
```

> /// \defgroup PkgFeatureGraphParameter Parameter Class

Reviewer: It seems that this group can be removed

The goal of the feature detection methods is to construct lines that can be used for future mesh generation.}
\cgalPkgManuals{Chapter_Feature_Graph,PkgFeatureGraphRef}
\cgalPkgSummaryEnd

\cgalPkgShortInfoBegin
\cgalPkgSince{6.2}
\cgalPkgBib{cgal:c-fg}
\cgalPkgLicense{\ref licensesGPL "GPL"}
\cgalPkgShortInfoEnd
\cgalPkgDescriptionEnd

The feature graph is extracted in multiple steps.
First, a sharpness evaluation is conducted on the surface elements.
Then, from this measure, smooth elements are discarded,
and a graph thinning process generates lines that
are guaranteed to be positioned along the highest values of the measure.
Finally, the result is refined with a correction step and an optimization phase
to ensure the correct placement of lines and corners along estimated normals.

\cgalClassifedRefPages

\cgalCRPSection{Concepts}

Concepts representing the input surfaces:

- `FeatureImage_3`
- `FaceListGraph`

Secondary concepts:

- `SharpnessEstimator`
- `NormalEstimator`

\cgalCRPSection{Feature graph creation functors}

- `CGAL::Feature_graph::Detect_sharp_features_on_labeled_image()`
- `CGAL::Feature_graph::Detect_sharp_features_on_surface()`

> \cgalCRPSection{Parameter classes}

Reviewer: ```suggestion
\cgalCRPSection{Parameter Classes}
```

- `CGAL::Feature_graph::Regularization_parameters_on_image`
- `CGAL::Feature_graph::Regularization_parameters_on_surface`
- `CGAL::Feature_graph::Optimization_parameters_on_image`
- `CGAL::Feature_graph::Optimization_parameters_on_surface`

> \cgalCRPSection{Sharpness estimators}

Reviewer: The next two lines show well a problem.  In the first line `sharpness` at the end, in the second line `sharpness`  in the middle.
And still `_functor` suffix which I suggested to avoid.

> \cgalCRPSection{Sharpness estimators}

Reviewer: ```suggestion
\cgalCRPSection{Sharpness Estimators}
```

- `CGAL::Feature_graph::AmbrosioTortorelli_on_image::Sharpness_functor`
- `CGAL::Feature_graph::Sharpness_estimator::Sharpness_estimator_on_surface`

> \cgalCRPSection{Normal estimators}

Reviewer: ```suggestion
\cgalCRPSection{Normal Estimators}
```
